### ISSUE TITLE
bokeh ipython magic imports plotting.hold, but plotting.hold is dead

### ISSUE THREAD

**Issue by jacksonloper** [bokeh/bokeh#1821]
Line 19 of 

https://raw.githubusercontent.com/bokeh/bokeh/master/extensions/bokeh_magic.py

says 

    from bokeh.plotting import (output_notebook, show, hold, figure)

However, AFAIK as of 9802a98d8137501f2e4a64c992424a7f09a20a4f, bokeh.plotting.hold no longer exists (it's not just deprecated, it's actually gone.)

Am I missing something obvious?  Happy Wednesday!

**Comment by damianavila** [comment]
Wow... not missing something obvious... we have to update the magic... well, I have to say that most of the functionality is probably not needed any more because of the new status of the plotting API, but I will take a look and see what is reasonable to keep or add... thanks for the report!

**Comment by damianavila** [comment]
Thinking on this... I am toward deprecating the magic and these are my reasons:

* %bokeh --hold does not have any sense now
* %bokeh --figure neither because you need to make `p = figure()` to actually show it later

So we have `%bokeh --notebook` to actually prepend `output_notebook` to each cell and `--show` to append `show(p)` at the end of each cell... but now `p` has to make reference to a `p=figure()` that we need to create at the beginning too, so now the user is in the very WEIRD situation of modification over a `p` object which is defined in the "eter"... but, further, what happens if the user wants to build `f=figure(with a lot of parameters here)`? OK, we can tell the user to use always `p` (which is something the user has to remember or we need to warn about it if it is not happening), so now they use `p` for the 1st cell and `p` for the second one, and then they want to build a gripplot or a widget... and obviously, we lost the reference for the first plot... so the plotting capabilities are limited by the magic itself.
In this scenerario I only see `output_notebook` in each cell as a valid request, and we can easily add this to plotting (or the cross API) with something like `output_notebook_each_cell` or something like that, and you are done. 
To summarize, I see some big "conceptual/abstraction" problems here, forcing the user to work with "magic" objects (although it is a `magic` :wink:). Maybe this is something for more advanced user in the notebook which want to do "dirty" a very simple plotting, but probably that proportion is minimal and I am worried about the common user of the Bokeh in the notebook: for them this magic could be very confusing...

Thoughts? ping @bokeh/dev

**Comment by fpliger** [comment]
I think you are right in your consideration. I'd very much avoid implicit things that are not clearly an obvious advantage for [almost all] users.

**Comment by damianavila** [comment]
@bryevdv I would like your thoughts here... thanks!

**Comment by bryevdv** [comment]
I think the main point of bokeh magic was to help make a stateful API more manageable in the notebook and now that we are moving away from the stateful API we can deprecate/remove bokeh magic as well.

**Comment by damianavila** [comment]
OK, I will deprecate it, just making it functional on previous version of bokeh but not working in the new ones... Then, after the release we can remove it completely.
